tst/com/amazon/ion/benchmark/DataGeneratorTest.java

```diff
@@ -189,13 +198,56 @@ public void testSizeOfGeneratedData() throws Exception {
         assertTrue(difference <= 0.1 * expectedSize);
     }
 
+
+    /**
+     * Test if there's violation by comparing every single data with the Ion schema constraint.
+     * @throws Exception if error occur
+     */
+    @Test
+    public void testViolation() throws Exception {
+        Map <String, Object> optionsMap = Main.parseArguments("generate", "--data-size", "5000", "--format", "ion_text", "--input-ion-schema", "./tst/com/amazon/ion/benchmark/testStruct.isl", "test8.ion");
+        String inputFilePath = optionsMap.get("--input-ion-schema").toString();
+        Path classSchemaFile = Paths.get(inputFilePath);
+        OUTPUT_FILE = optionsMap.get("<output_file>").toString();
+        try (
+                IonReader readerInput = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(inputFilePath)));
+                IonReader reader = DataGeneratorTest.executeAndRead(optionsMap);
+                Stream<String> lines = Files.lines(classSchemaFile)
+        ) {
+            // Get the name of Ion Schema.
+            IonDatagram schema = ReadGeneralConstraints.LOADER.load(readerInput);
+            String ionSchemaName = null;
+            for (int i = 0; i < schema.size(); i++) {
+                IonValue schemaValue = schema.get(i);
+                if (schemaValue.getType().equals(IonType.STRUCT) && schemaValue.getTypeAnnotations()[0].equals(IonSchemaUtilities.KEYWORD_TYPE)) {
+                    IonStruct constraintStruct = (IonStruct) schemaValue;
+                    ionSchemaName = constraintStruct.get(IonSchemaUtilities.KEYWORD_NAME).toString();
+                    break;
+                }
+            }
+            // Construct new schema amd get the type of the Ion Schema.
+            String content = lines.collect(Collectors.joining(System.lineSeparator()));
+            Schema newSchema = ISS.newSchema(content);
+            Type type = newSchema.getType(ionSchemaName);
+            while (reader.next() != null) {
+                IonValue value = SYSTEM.newValue(reader);
+                Violations violations = type.validate(value);
+                if (!violations.isValid()) {
+                    System.out.println(value);
+                    System.out.println(violations);
+                }
+                assertTrue(violations.isValid());
```

Comment:
> Instead of using explicit `System.out.println`s, I recommend something like
> 
> ```
> assertTrue("Violations " + violations + "found in value " + value, violations.isValid());
> ```

Change has been updated in the following commit